packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -78,23 +78,27 @@ export const start = () => {
     }
 
     const error = window.__ERROR__
-
     return Promise.resolve()
         .then(() => new Promise((resolve) => loadableReady(resolve)))
         .then(() => {
             ReactDOM.hydrate(
                 <ExpressContext.Provider value={{}}>
                     <Router>
-                        <DeviceContext.Provider value={{type: window.__DEVICE_TYPE__}}>
-                            <AppConfig locals={locals}>
-                                <Switch
-                                    error={error}
-                                    appState={window.__PRELOADED_STATE__}
-                                    routes={routes}
-                                    App={WrappedApp}
-                                />
-                            </AppConfig>
-                        </DeviceContext.Provider>
+                        <CorrelationIdProvider
+                            correlationId={uuidv4()}
+                            onPageChange={() => uuidv4()}
```

Comment:
Ok.. so last comment about this API. This is mainly feedback on my own suggestions. But I was a little wary about the name 'onPageChange' with respect to how it assigns the return value to the id. It's not really that self explanatory is it?

```
<CorrelationIdProvider id={string | () => string} resetOnPageChange={boolean} />
```

Above is another suggestion that feels slightly better. The id can either be a string or a function that returns a string. Then we have the boolean prop resetOnPageChange that if set to true will invoke the id function again. 

I think that is a little cleaner. One caveat is that we might want to warn if the dev is passing a string id and true for resetting.